### PR TITLE
Added possibility for health checks to time out and run in parallel

### PR DESCRIPTION
**Pull request by tosan88**
Following standards the default for __health should be 10s timeout, for __gtg 3s

### CODE REVIEW

File: src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java [hunk 1]
```diff
@@ -2,21 +2,44 @@
...
  */
 public class DefaultGoodToGoChecker implements GoodToGoChecker {
 
-	@Override
-	public boolean isGoodToGo(Environment environment) {
-		for (AdvancedResult result : HealthChecks.runAdvancedHealthChecksIn(environment).values()) {
-			if (result.status() == AdvancedResult.Status.ERROR) {
-				return false;
-			}
+	private static final int DEFAULT_TIMEOUT = 3;
```
Comment (tamas-molnar) [inline src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java]: timeout unit?

File: src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java [hunk 2]
```diff
@@ -2,21 +2,44 @@
...
+			return new GoodToGoResult(executorService.submit(() -> {
+				for (AdvancedResult result : HealthChecks.runAdvancedHealthChecksIn(environment).values()) {
+					if (result.status() == AdvancedResult.Status.ERROR) {
+						return false;
+					}
+				}
+				return true;
+			}).get(this.timeOut, TimeUnit.SECONDS), "");
+		} catch (InterruptedException | ExecutionException | TimeoutException e) {
+			return new GoodToGoResult(false, "Timeout running status check");
```
Comment (tamas-molnar) [inline src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java]: Consider including the timeout value in the message.

File: src/main/java/com/ft/platform/dropwizard/AdvancedSynchronousHealthChecksRunner.java [hunk 3]
```diff
@@ -10,45 +10,46 @@
 
 public class AdvancedSynchronousHealthChecksRunner {
 
+    private static final int DEFAULT_TIMEOUT = 10;
```
[the remark this answers is not in the document]
Comment (tosan88) [reply src/main/java/com/ft/platform/dropwizard/AdvancedSynchronousHealthChecksRunner.java]: Will add in the name

File: src/main/java/com/ft/platform/dropwizard/AdvancedSynchronousHealthChecksRunner.java [hunk 4]
```diff
@@ -10,45 +10,46 @@
...
     private final Environment environment;
     private final ObjectMapper objectMapper;
     private final String appName;
     private final String appDescription;
     private final HealthcheckFailureReporter healthcheckFailureReporter;
     private final String systemCode;
+    private final int timeOut;
```
[the remark this answers is not in the document]
Comment (tosan88) [reply src/main/java/com/ft/platform/dropwizard/AdvancedSynchronousHealthChecksRunner.java]: Will add in the name

File: src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java (hunk 1 above)
Comment (tosan88) [reply src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java]: Will add in the name

File: src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java (hunk 2 above)
Comment (tosan88) [reply src/main/java/com/ft/platform/dropwizard/DefaultGoodToGoChecker.java]: Good point

**Comment by tosan88** [comment]
@tamas-molnar Fixed and additionally removed some old configuration (FYI)

**Pull request merged**